### PR TITLE
APPSRE-6321 Add ShardSpecOverride to MR checks

### PR DESCRIPTION
**Pull request by janboll**
Make it possible to override the image for Merge Request checks

### CODE REVIEW

File: schemas/app-sre/integration-1.yml [hunk 1]
```diff
@@ -55,6 +55,24 @@ properties:
...
+              enum:
+                - per-aws-account
+            awsAccount:
+              "$ref": "/common-1.json#/definitions/crossref"
+              "$schemaRef": "/aws/account-1.yml"
+            imageRef:
+              type: string
+          required:
+          - awsAccount
+          - imageRef
```
Comment (maorfr) [inline schemas/app-sre/integration-1.yml:75]: if this is now a duplicate, i would suggest extracting it to a separate schema

Comment (janboll) [reply schemas/app-sre/integration-1.yml:75]: Good point! Will do

Comment (janboll) [reply schemas/app-sre/integration-1.yml]: I refactored the schema. However this now crashes the qontract-server. Validation works fine though.

Comment (janboll) [reply schemas/app-sre/integration-1.yml]: Nevermind, was related to local image mess

**Review by cubismod — APPROVED** [review]

File: schemas/app-sre/integration-shardspec-override-1.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      - per-aws-account
+  awsAccount:
+    "$ref": "/common-1.json#/definitions/crossref"
+    "$schemaRef": "/aws/account-1.yml"
+  imageRef:
+    type: string
+
+required:
+- awsAccount
+- imageRef
```
Comment (geoberle) [inline schemas/app-sre/integration-shardspec-override-1.yml:26]: no end of line 😱

**Review by geoberle — APPROVED** [review]
nit - otherwise lgtm

File: schemas/app-sre/integration-shardspec-override-1.yml (hunk 2 above)
Comment (janboll) [reply schemas/app-sre/integration-shardspec-override-1.yml]: added

**Pull request merged**